lib/dust.js

```diff
@@ -792,6 +807,20 @@
     }
   };
 
+  Chunk.prototype.thenable = function(thenable, context, body) {
+    return this.map(function(chunk) {
+      thenable.then(function(data) {
+        if(body) {
+          chunk.render(body, context.push(data)).end();
+        } else {
+          chunk.end(data);
+        }
+      }, function(err) {
+        chunk.setError(err);
```

Comment:
I like the `{:error}` idea. Let's make sure this gets into the documentation before/as this feature gets released.